### PR TITLE
Update playingEffectPromise synchronously

### LINKED ISSUE #179: GamepadHapticActuator.playEffect() algorithm seems wrong with regards to `[[playingEffectPromise]]`
GamepadHapticActuator.playEffect() algorithm seems wrong with regards to `[[playingEffectPromise]]`:
- https://w3c.github.io/gamepad/extensions.html#dom-gamepadhapticactuator-playeffect

Step 4 says to *queue a task* to set `[[playingEffectPromise]]` to undefined. Step 6 says to let `[[playingEffectPromise]]` be a new promise. I believe The queued in task 4 would thus clear the new promise instead of the one from the previously playing effect.

### PR DESCRIPTION
**Pull request by nondebug**
Closes #179

The following tasks have been completed:

 * [ ] Modified Web platform tests (link to pull request)

Implementation commitment:

 * [ ] WebKit (https://bugs.webkit.org/show_bug.cgi?id=)
 * [ ] Chromium (https://bugs.chromium.org/p/chromium/issues/detail?id=)
 * [ ] Gecko (https://bugzilla.mozilla.org/show_bug.cgi?id=)

### CODE REVIEW

File: extensions.html [hunk 1]
```diff
@@ -277,18 +277,19 @@ <h2>
...
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}.
                 </li>
                 <li>Set
                 [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} to
                 `undefined`.
                 </li>
+                <li>[=Queue a global task=] on the [=relevant global object=]
+                of [=this=] using the [=gamepad task source=] to [=resolve=]
+                |effectPromise| with {{GamepadHapticsResult/"preempted"}}.
+                </li>
```
Comment (marcoscaceres) [inline extensions.html:292]: This code path ends with promise being resolved, right? 

```suggestion
                </li>
                <li>Terminate this algorithm.</li>
```

File: extensions.html [hunk 2]
```diff
@@ -354,18 +355,19 @@ <h2>
...
```
Comment (marcoscaceres) [inline extensions.html:370]: As above?? 

```suggestion
                </li>
                <li>Terminate this algorithm.</li>
```

**Review by marcoscaceres — APPROVED** [review]

File: extensions.html (hunk 1 above)
Comment (nondebug) [reply extensions.html:292]: This path is for preempting the currently playing effect to prepare for the new effect. So there are two promises, the old effect promise and the new effect promise. We want to resolve the first promise (if it exists) and then create the new one. If we terminate here then we won't start the new effect.

File: extensions.html (hunk 2 above)
Comment (nondebug) [reply extensions.html:370]: If we terminate here then we won't stop vibration when there's an effect playing.

**Pull request merged**